containers-tests/benchmarks/Set.hs

```diff
@@ -45,6 +45,8 @@ main = do
         , bench "alterF:four:strings" $ whnf (alterF_four strings) strings_s
         , bench "alterF_naive:four" $ whnf (alterF_naive_four elems) s
         , bench "alterF_naive:four:strings" $ whnf (alterF_naive_four strings) strings_s
+        , bench "powerSet" $ whnf S.powerSet (S.fromList[1..20])
+        , bench "member.powerSet" $ whnf (\ s -> all (flip S.member s) s) (S.powerSet (S.fromList [1..20]))
```

Comment:
20 seems a bit large, and if want to check both odd and even sizes.

Reply:
I was thinking to check on the "alignment" of subset size and depth in different circumstances. But now that I think about it, it's probably better to get rid of the lookup benchmark and instead add a property test checking that. Do you think you can formulate one?

Reply:
I wrote that thinking only about correctness, and not performance. If we can formulate a shape property that will ensure good performance for random lookups, I think that would be valuable too. If someone later finds a substantially faster construction that violates that property, we can evaluate how bad it really is. Checking shape and sizes is a lot cheaper than inspecting the actual contents of each subset, which is what my original property tests used to do (I guess that wasn't my smartest day).

Reply:
I'm fine with having the benchmark, and I won't hold things up for the property test. That said, I don't promise not to add one in the future. While we can't say for certain that a particular property will give better performance for all/any applications, it seems clear that both tree balance and the arrangement of element sizes will have *some* impact on performance, so future changes that modify those should require some thought into whether that's likely a good or bad change. I agree that pure lookups into a power set are silly; it would pretty much always make more sense to keep the underlying set around and do a subset query against that. But a power set that's been modified by relatively few insertions and/or deletions will maintain much of the shape it started with.